Add repair logic for broken indexes

nathanielc opened the pull request:
Fixes #1295 

Also Fixes #507 since using any of these operations is potentially destructive we needed a backup endpoint.

- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x]  Migrate the repair process to be an API call instead of a startup action.

nathanielc commented on services/replay/service.go:
```diff
@@ -120,6 +120,12 @@ func (s *Service) Open() error {
 
+	//TODO: Should we expose this via the API instead of doing it during every startup?
+	// This could be expensive and should not need to be run frequently.
+	if err := s.repairRecordingIndex(); err != nil {
```
@desa What do you think we should do here?

desa replied:
I'd say have it be a part of the API and not to run it on startup.

desa replied:
Sorry that I missed this earlier.

nathanielc commented:
@desa Could you give this another look now that I have exposed the rebuild process through the API?

nathanielc commented:
@desa Actually if you want to hold off, I just realized I need to make a change.

nathanielc commented:
@desa OK this is ready for review. Summary of changes since your first review:

* Added API endpoint to create database Backups
* Added API endpoints to perform actions on various underlying stores.

desa commented on client/API.md:
````diff
@@ -2116,6 +2117,62 @@ POST /kapacitor/v1/config/influxdb/remote
+
+```
+# Create a backup.
+curl http://localhost:9092/kapacitor/v1/storage/backup > kapacitor.db
````
maybe we should have a `kapacitor backup` sub-command?

nathanielc replied:
Agreed, I'll put this together. I don't want a subcommand for the rebuild action though, I figure those are best only in the API so that you have to read and be aware of how they work.

desa replied:
Should we update the example to use the backup subcommand?

nathanielc replied:
I think the example here using the API is fine because it is part of the API docs.

desa replied:
kk

desa commented:
Just Finished reading. From what I can tell LGTM.
